pkg/cli/alpha/internal/update/update.go

```diff
@@ -0,0 +1,372 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package update
+
+import (
+	"errors"
+	"fmt"
+	"io"
+	"net/http"
+	"os"
+	"os/exec"
+	"time"
+
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+	"sigs.k8s.io/kubebuilder/v4/pkg/plugin/util"
+)
+
+// Update contains configuration for the update operation
+type Update struct {
+	// FromVersion stores the version to update from, e.g., "v4.5.0".
+	FromVersion string
+	// ToVersion stores the version to update to, e.g., "v4.6.0".
+	ToVersion string
+	// FromBranch stores the branch to update from, e.g., "main".
+	FromBranch string
+
+	// UpdateBranches
+	AncestorBranch string
+	OriginalBranch string
+	UpgradeBranch  string
+	MergeBranch    string
+}
+
+// Update a project using a default three-way Git merge.
+// This helps apply new scaffolding changes while preserving custom code.
+func (opts *Update) Update() error {
+	log.Infof("Checking out base branch: %s", opts.FromBranch)
+	checkoutCmd := exec.Command("git", "checkout", opts.FromBranch)
+	if err := checkoutCmd.Run(); err != nil {
+		return fmt.Errorf("failed to checkout base branch %s: %w", opts.FromBranch, err)
+	}
+
+	suffix := time.Now().Format("02-01-06-15-04")
+
+	opts.AncestorBranch = "tmp-ancestor-" + suffix
+	opts.OriginalBranch = "tmp-original-" + suffix
+	opts.UpgradeBranch = "tmp-upgrade-" + suffix
+	opts.MergeBranch = "tmp-merge-" + suffix
+
+	log.Infof("Using branch names:")
+	log.Infof("  Ancestor: %s", opts.AncestorBranch)
+	log.Infof("  Original:  %s", opts.OriginalBranch)
+	log.Infof("  Upgrade:  %s", opts.UpgradeBranch)
+	log.Infof("  Merge:    %s", opts.MergeBranch)
+
+	// 1. Creates an ancestor branch based on base branch
```

Comment:
Let's add the macro log steps here so that we can remove ALL info logs inside of the functions called and keep things cleaner.